staging/src/k8s.io/component-base/metrics/prometheus/restclient/metrics.go

```diff
@@ -54,13 +54,53 @@ var (
 		},
 		[]string{"code", "method", "host"},
 	)
+
+	execPluginCertExpiration = k8smetrics.NewGauge(
+		&k8smetrics.GaugeOpts{
+			Name: "rest_client_exec_plugin_certificate_expiration_seconds",
+			Help: "Gauge of the nearest expiration of the client " +
+				"certificate(s) managed by the auth exec plugin. The value " +
+				"is in seconds since January 1, 1970 UTC. If auth exec " +
```

Comment:
@logicalhan are time-based metrics like this normal? it means that a "healthy" value for this metric would change over time, right? would a ttl in seconds be more useful or easier to set up alerts for (e.g. "alert when rest_client_exec_plugin_certificate_expiration_seconds is not 0 and is less than 259200" or something)?

Reply:
or is it normal to have time values in a gauge and metrics inspectors can derive the ttl at a point in time from that?

Reply:
> Han Kang are time-based metrics like this normal? it means that what is considered "healthy" for this metric would change over time, right? would a ttl in seconds be more useful or easier to set up alerts for (e.g. "alert when rest_client_exec_plugin_certificate_expiration_seconds is not 0 and is less than 259200" or something)?

Using TTL in seconds would be better.

Reply:
Done, it now reports the shortest TTL on every getCreds().